### PR TITLE
rhel related monitoring change

### PR DESCRIPTION
**Pull request by Xiaohua-Shen**
1. as https://github.com/blueboxgroup/ursula-monitoring/pull/120  sam cooper's suggestion,  I move the ruby path from sensu-plugin to sensu user's path.

2.  will add rhel related monitoring change after https://github.com/blueboxgroup/ursula/pull/2363 merged.
-- openstack_install_method == distro, no need to add external sensu repo;
-- openstack_install_method == distro,  use /etc/sysconfig/sensu to replace /etc/default/sensu
-- openstack_install_method == distro, change openstack related service's process monitoring, like nova-api change to openstack-nova-api

### CODE REVIEW

**Comment by nirajdp76** [comment]
ok to test

**Comment by scoop206** [comment]
test this please

**Comment by epwn** [comment]
retest

**Comment by epwn** [comment]
retest

**Comment by Xiaohua-Shen** [comment]
retest

**Comment by Xiaohua-Shen** [comment]
@scoop206 @paulczar please review this rhosp related PR:
    -- openstack_install_method == distro, no need to add external sensu repo;
    -- openstack_install_method == distro, use /etc/sysconfig/sensu to replace /etc/default/sensu
    -- openstack_install_method == distro, change openstack related service's process monitoring, like nova-api change to openstack-nova-api
    -- openstack_install_method == distro, add missing rubygem
    -- openstack_install_method == distro, since no sensu embedded ruby and no keystone related virtualenv, add symbolic link for compatible

File: roles/common/tasks/monitoring.yml [hunk 1]
```diff
@@ -50,18 +50,57 @@
...
+    shell: "{{ ruby_path }}gem install {{ item.name }} --no-user-install {{ '-v ' if item.version is defined else ' ' }} {{ item.version|default(' ') }}"
+    register: result
+    until: result|succeeded
+    retries: 5
+    with_items: "{{ monitoring.checks.gems_rhel_distro }}"
+
+  - name: create path for sensu embedded ruby
+    file: dest=/opt/sensu/embedded/bin state=directory owner=sensu mode=0750
+
+  - name: add symbolic link for embedded ruby path (to compatible with some sensu-plugins's hardcode embeded ruby path)
```
Comment (scoop206) [inline roles/common/tasks/monitoring.yml]: now that we're setting PATH in /etc/default/sensu on Ubuntu can we instead remove the hardcoded /opt/sensu/embedded/bin/ruby from all the shebangs in the checks  ?
Looking them over and it seems feasible.

File: roles/common/tasks/monitoring.yml [hunk 2]
```diff
@@ -50,18 +50,57 @@
...
+    file: src='/usr/bin/ruby' dest='/opt/sensu/embedded/bin/ruby' state=link
+
+  - name: add symbolic link for embedded gem path (to compatible with some hardcode gem install)
```
Comment (scoop206) [inline roles/common/tasks/monitoring.yml]: we need this for serverspec ?

**Comment by Xiaohua-Shen** [comment]
retest

**Comment by Xiaohua-Shen** [comment]
as @scoop206  requested, we still use sensu community package, so I removed related change in the commit, also I pick up the commit from #2376 in this PR too. please review again.

**Comment by paulczar** [comment]
this is looking pretty good,  is it ready?  or still WIP ?

**Comment by Xiaohua-Shen** [comment]
@paulczar, it's ready, please merge it . thanks.

**Pull request merged**